src/components/fields/Array/index.tsx

```diff
@@ -0,0 +1,33 @@
+import { lazy } from "react";
+import DataArrayIcon from "@mui/icons-material/DataArray";
+
+import { IFieldConfig, FieldType } from "@src/components/fields/types";
+import withRenderTableCell from "@src/components/Table/TableCell/withRenderTableCell";
+
+import DisplayCell from "./DisplayCell";
+
+const SideDrawerField = lazy(
+  () =>
+    import("./SideDrawerField" /* webpackChunkName: "SideDrawerField-Array" */)
+);
+const Settings = lazy(
+  () => import("./Settings" /* webpackChunkName: "Settings-Array" */)
+);
+export const config: IFieldConfig = {
+  type: FieldType.array,
+  name: "Array",
+  group: "Code",
+  dataType: "object",
+  initialValue: undefined,
+  initializable: true,
+  icon: <DataArrayIcon />,
+  settings: Settings,
+  description:
+    "Connects to a sub-table in the current row. Also displays number of rows inside the sub-table. Max sub-table depth: 100.",
+  TableCell: withRenderTableCell(DisplayCell, SideDrawerField, "popover", {
+    popoverProps: { PaperProps: { sx: { p: 1, minWidth: "200px" } } },
```

Comment:
I went with the popover approach because the sub-table approach opens a Modal from the bottom, and thus the user loses the context of the row since he has to close the modal to see the row and the Modal breaks the flow, IMO Popover approach seems to be simple steps to go with. Let me know I can use modal in case this isn't what we want

Reply:
You can try embedding the table component inside the popover, however providing a way to configure the column schema might be more challenging